core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -783,6 +817,38 @@ private[spark] class TaskSetManager(
             s"Lost task ${info.id} in stage ${taskSet.id} (TID $tid) on ${info.host}, executor" +
               s" ${info.executorId}: ${ef.className} (${ef.description}) [duplicate $dupCount]")
         }
+
+        // for next round increase cpusPerTask for OOME/LME
+        if (supportsDynamicCpusPerTask && !isZombie && (ef.className.contains("OutOfMemory") ||
```

Comment:
ExceptionFailure will have LME or OOM always in the classname or you want to handle the cause areas as well in the deep stacktrace? or that is not relevant here?

Reply:
Haven't seen deeper in the stacktrace so far. Might be safer to add that so will do.